CMakeLists.txt

```diff
@@ -1,56 +1,67 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(teleop_twist_joy)
 
-find_package(catkin REQUIRED COMPONENTS geometry_msgs roscpp sensor_msgs)
+if(NOT WIN32)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -Wall -Wextra -Wpedantic")
+endif()
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES ${PROJECT_NAME}
-  CATKIN_DEPENDS geometry_msgs roscpp sensor_msgs
-)
+find_package(ament_cmake_ros REQUIRED)
+
+find_package(geometry_msgs REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(sensor_msgs REQUIRED)
 
-include_directories(include ${catkin_INCLUDE_DIRS})
+include_directories(include)
 
 add_library(${PROJECT_NAME} src/${PROJECT_NAME})
-target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES})
+
+ament_target_dependencies(${PROJECT_NAME}
+  "geometry_msgs"
+  "rclcpp"
+  "sensor_msgs"
+)
 
 add_executable(${PROJECT_NAME}_node src/teleop_node.cpp)
```

Comment:
@mikaelarguedas @dirk-thomas Just stumbling across this now via the ROSCon talk! Just a quick question— couldn't this wrapper executable have been dropped? My impression is that ROS2 provides this capability natively so that the package could just supply a single library that can be used either as a standalone process or nodelet-style.

Reply:
Currently we don't have nice to use command line tools to instantiate a node based on the shared library. (There is only a demo demonstrating the technology at the moment.) In the future this executable should indeed be obsolete.

Reply:
Cool.